runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java

```diff
@@ -156,6 +171,67 @@
 
 	private GrandExchangeClient grandExchangeClient;
 
+	private boolean wasFuzzySearch;
+
+	/**
+	 * Logic from {@link org.apache.commons.text.similarity.FuzzyScore}
+	 */
+	private static List<Integer> findFuzzyIndices(String term, String query)
+	{
+		List<Integer> indices = new ArrayList<>();
+
+		// fuzzy logic is case insensitive. We normalize the Strings to lower
+		// case right from the start. Turning characters to lower case
+		// via Character.toLowerCase(char) is unfortunately insufficient
+		// as it does not accept a locale.
+		final String termLowerCase = term.toLowerCase();
+		final String queryLowerCase = query.toLowerCase();
+
+		// the position in the term which will be scanned next for potential
+		// query character matches
+		int termIndex = 0;
+
+		for (int queryIndex = 0; queryIndex < queryLowerCase.length(); queryIndex++)
+		{
+			final char queryChar = queryLowerCase.charAt(queryIndex);
+
+			boolean termCharacterMatchFound = false;
+			for (; termIndex < termLowerCase.length()
+					&& !termCharacterMatchFound; termIndex++)
+			{
+				final char termChar = termLowerCase.charAt(termIndex);
+
+				if (queryChar == termChar)
+				{
+					indices.add(termIndex);
+
+					// we can leave the nested loop. Every character in the
+					// query can match at most one character in the term.
+					termCharacterMatchFound = true;
+				}
+			}
```

Comment:
`termCharacterMatchFound` can be removed and be replaced by a `break;` statement.
```suggestion
			while (termIndex < termLowerCase.length())
			{
				final char termChar = termLowerCase.charAt(termIndex);

				if (queryChar == termChar)
				{
					indices.add(termIndex);

					// we can leave the nested loop. Every character in the
					// query can match at most one character in the term.
					break;
				}
				
				termIndex++;
			}
```